Library/Homebrew/formula.rb

```diff
@@ -502,6 +508,21 @@ def installed_kegs
     installed_prefixes.map { |dir| Keg.new(dir) }
   end
 
+  # Formula reference to use to get an identical installation of the formula.
+  #
+  # Usually, the formula's path is a good canonical identifier for the formula.
+  # Just using whatever was passed to `brew install` isn't a good idea, because
+  # if it was e.g. a URL, it could end up being downloaded twice.
+  #
+  # However, if the formula was installed with an alias (i.e. `install_name` is
+  # different from `name`), that should be used instead so that information is
+  # preserved. Aliases are looked up in repositories that have already been
+  # tapped, so we don't have to worry about doing extra HTTP requests, or other
+  # expensive operations, when looking them up again.
+  def install_ref
+    install_name == name ? path : install_name
```

Comment:
Didn't occur to me that I could return the path to the alias, but it's obvious in hindsight!